### PR TITLE
Allow support for PHPUnit 8.x

### PR DESCRIPTION
**Pull request by bennothommo**
For your consideration, this refactors this library to work with PHPUnit 8.x. As PHPUnit 8 was a large refactor on its own, I opted to make this a complete refactor of the library as opposed trying to maintain support for the older versions of PHPUnit, so I would recommend this be released as a new major version.

Please let me know if you would like me to make any further adjustments in order for this to be considered for merging.

### CODE REVIEW

**Comment by meyfa** [comment]
Thanks for your contribution, I see you took care of everything, no changes needed.

I've kept the compatibility for so long because this library was written to test another project of mine, php-svg, that makes a lot more sense with PHP 5 support.

I'll create a 1.x branch and report back shortly.

**Comment by meyfa** [comment]
All good! Merging, and will create a 2.0 release.

**Comment by bennothommo** [comment]
@meyfa Thank you very much for merging, and thanks heaps for the great PHPUnit plugin :)

File: composer.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
     ],
     "require": {
         "php": ">=7.2",
-        "phpunit/phpunit": "^8.0|^9.0",
```
Comment (iotch) [inline composer.json]: Is there any reason to remove support of phpunit v9?

**Pull request merged**